### DISCUSSION TITLE
DataPortal Methods and DateTimeOffset Requirement

### DISCUSSION THREAD

**Discussion by skeeler88** [MarimerLLC/csla#2522] [Questions] [answered]
Hi there,

I am curious as to a generic way or what best practice does anyone take whilst having the need to pass a DateTimeOffset to a DataPortal method. Because this type is not serializable, my only thought is to pass a string and parse that value to a DateTimeOffset on the server. Though quite labor intensive considering an app may have these parameters littered throughout, I am hoping for maybe a different answer. The solution I mentioned also does not restrict nor accurately describe the the info needed for those/that parameter(s).

What is the common practice to pass this type of information & to keep it serializable? Or is the above mentioned solution the recommended way? And if it is not in the correct format just expect a DataPortal exception?

e.g.

`[Fetch]
 private async Task Fetch(DateTimeOffset dateFrom, DateTimeOffset dateTo)` and as expected results in: 

`
Warning	CSLA0010	Operation argument types should be serializable
`

Thank you.

**Comment by rockfordlhotka** [comment, marked as answer]
There is some discussion of generalized solutions for having MobileFormatter serialize otherwise nonserializable types in this thread: https://github.com/MarimerLLC/csla/discussions/2386.

For something like DateTimeOffset, it is probably worth enhancing MobileFormatter to directly understand the type so it becomes serializable via MobileFormatter.

**Reply by rockfordlhotka** [reply]
Added to the backlog: https://github.com/MarimerLLC/csla/issues/2523
